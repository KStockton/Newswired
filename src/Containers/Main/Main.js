import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchAllBooks } from '../../Thunks/fetchAllBooks';
import { fetchTopTravel } from '../../Thunks/fetchTopTravel';
import BooksContainer from '../../Containers/BooksContainer/BooksContainer';
import TravelContainer from '../../Containers/TravelContainer/TravelContainer';
import Loading from '../../Components/Loading/Loading';
import Nav from '../Nav/Nav';
const API_KEY =`${process.env.REACT_APP_NEWYORKTIMES_API_KEY}`

// import { cleanBooks } from '../../Utility/cleanBooks';



class Main extends Component {
  

  async componentDidMount() {

    

    if(this.props.allBooks.length || this.props.allTravel.length){
      // //prevents a second fetch when going back and forth
    }
    else {
      // const travelUrl = `https://api.nytimes.com/svc/topstories/v2/travel.json?api-key=${API_KEY}`
      // this.props.fetchTopTravel(travelUrl)
      
      const bookUrl = `https://api.nytimes.com/svc/books/v3/lists/current/hardcover-fiction.json?api-key=${API_KEY}`
      this.props.fetchAllBooks(bookUrl)
      }
  }



  render() {
    return (
      <div className="">
        <div className="Main-header">
          <h1 className="Main-title">News Wire</h1>
          <Nav/>
        </div>
        {
          (!this.props.isLoading) ? 
          // <TravelContainer travel={this.props.allTravel}/>
          <BooksContainer books={this.props.allBooks}/>
           :
            <Loading/>
        }
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
 allBooks: store.allBooks,
 allTravel: store.allTravel,
 isLoading: store.isLoading
})

export const mapDispatchToProps = (dispatch) => ({
fetchAllBooks: (url) => dispatch(fetchAllBooks(url)),
fetchTopTravel: (url) => dispatch(fetchTopTravel(url))
})



export default connect(mapStateToProps, mapDispatchToProps) (Main)