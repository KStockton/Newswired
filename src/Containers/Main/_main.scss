.Main-background{
  // background-image: url('../Assets/mainbg.jpg');
  // background-repeat: no-repeat;
  // background-size: 100% 100%;
  // background-position: center;
  // height: 100vh;
  // width: 100vw;
}


.Main-header {
background-color: #ccd0d56b;
margin: 0;
width: 100vw;
height: 5rem;
display: flex;
align-items: center;
justify-content: space-between;
border-bottom: 2px double grey;

  .Main-title {
    margin: 0;
    font-family: 'Old Standard TT', serif;
    text-align: center;
    font-size: 2rem;
    padding-top: 8px;
    padding-left: 15px;
  }
}

.Main-fontawesome {
  font-size: 30px;
  padding-right: 8px;
}

.news-title{
  text-decoration: none;
  color: inherit;
  cursor: pointer;
  
  &:hover {
    color: white
  }
}