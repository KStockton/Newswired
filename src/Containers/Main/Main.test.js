import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';
import { Main, mapStateToProps} from './Main';

describe('Main', () => {
  let wrapper;
  let mockcategories = {books: true, travel: true, topNews: true, sportsNews: true};
  
  beforeEach(() => {
    wrapper = shallow(<Main categories={mockcategories}/>);
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Clicking on NewsWire should take you back to /Options', () => {
    expect(wrapper.find(Link).props().to).toBe('/Options');
  });

  describe('MSTP', () =>{

    it('should map state from mockstate', () => {
      const mockState ={ categories:{ books: true, travel: true, topNews: true, sportsNews: true}};
      const result = mapStateToProps(mockState);
      expect(result).toEqual(mockState);
    });
  });
});