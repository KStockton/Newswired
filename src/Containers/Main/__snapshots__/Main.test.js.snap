// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match the snapshot 1`] = `
<div
  className=""
>
  <div
    className="Main-header"
  >
    <Link
      className="news-title"
      to="/Options"
    >
      <h1
        className="Main-title"
      >
        News Wire
      </h1>
    </Link>
    <Nav />
  </div>
  <Connect(BooksContainer) />
  <Connect(TravelContainer) />
  <Connect(SportContainer) />
  <Connect(TopNewsContainer) />
</div>
`;
